src/runner/predictors/base_predictor.py

```diff
@@ -2,135 +2,72 @@
-        elif isinstance(batch, tuple):
-            return tuple(self._allocate_data(data) for data in batch)
-        elif isinstance(batch, torch.Tensor):
-            return batch.to(self.device)
-
-    def _get_inputs_targets(self, batch):
-        """Specify the data inputs and targets.
-        Args:
-            batch (dict or sequence): A batch of data.
-
-        Returns:
-            inputs (torch.Tensor or sequence of torch.Tensor): The data inputs.
-            targets (torch.Tensor or sequence of torch.Tensor): The data targets.
+            test_dict (dict): The computed results.
+                test_dict['losses'] (dict)
+                test_dict['metrics'] (dict, optional)
         """
         raise NotImplementedError
 
-    def _compute_losses(self, outputs, targets):
-        """Compute the losses.
-        Args:
-            outputs (torch.Tensor or sequence of torch.Tensor): The model outputs.
-            targets (torch.Tensor or sequence of torch.Tensor): The data targets.
-
-        Returns:
-            losses (sequence of torch.Tensor): The computed losses.
-        """
-        raise NotImplementedError
-
-    def _compute_metrics(self, outputs, targets):
-        """Compute the metrics.
-        Args:
-            outputs (torch.Tensor or sequence of torch.Tensor): The model outputs.
-            targets (torch.Tensor or sequence of torch.Tensor): The data targets.
-
-        Returns:
-            metrics (sequence of torch.Tensor): The computed metrics.
-        """
-        raise NotImplementedError
-
-    def _init_log(self):
-        """Initialize the log.
-        Returns:
-            log (dict): The initialized log.
-        """
-        log = {}
-        log['Loss'] = 0
-        for loss_fn in self.loss_fns:
-            log[loss_fn.__class__.__name__] = 0
-        for metric_fn in self.metric_fns:
-            log[metric_fn.__class__.__name__] = 0
-        return log
-
-    def _update_log(self, log, batch_size, loss, losses, metrics):
-        """Update the log.
-        Args:
-            log (dict): The log to be updated.
-            batch_size (int): The batch size.
-            loss (torch.Tensor): The weighted sum of the computed losses.
-            losses (sequence of torch.Tensor): The computed losses.
-            metrics (sequence of torch.Tensor): The computed metrics.
-        """
-        log['Loss'] += loss.item() * batch_size
-        for loss_fn, loss in zip(self.loss_fns, losses):
-            log[loss_fn.__class__.__name__] += loss.item() * batch_size
-        for metric_fn, metric in zip(self.metric_fns, metrics):
-            log[metric_fn.__class__.__name__] += metric.item() * batch_size
-
     def load(self, path):
         """Load the model checkpoint.
         Args:
             path (Path): The path to load the model checkpoint.
         """
-        checkpoint = torch.load(path, map_location=self.device)
+        checkpoint = torch.load(path, map_location='cpu')
```

Comment:
Confused.

Reply:
Just for the consistency. Please refer to [L255 in base_trainer.py](https://github.com/cmlab-mira/template/blob/refactor/src/runner/trainers/base_trainer.py#L255). The reason is that `checkpoint['torch_random_state']` and `checkpoint['torch_cuda_random_state']` should be loaded as `torch.Tensor`, not `torch.CudaTensor`.

Reply:
I see.